File: spring-boot-2/src/main/java/com/example/configproperties/TestController.java
package com.example.configproperties;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TestController {

    private final MappedProperties properties;

    public TestController(MappedProperties properties) {
        this.properties = properties;
    }

    @GetMapping("/test")
    public String get() {
        return properties.getFoo() + " " + properties.getBar();
    }

}

File: kotlin/spring-boot-2.0/src/main/kotlin/com/example/TestController.kt
package com.example

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class TestController(
        private val properties: MappedProperties
) {

    @GetMapping("/test")
    fun get(): String {
        return properties.foo.xur + " " + properties.bar
    }

}
